dom/src/main/scala/fs2/dom/MutationObserver.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2022 Arman Bilge
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2.dom
+
+import cats.syntax.all._
+import org.scalajs.dom
+import cats.effect.kernel.Async
+import cats.effect.std.Dispatcher
+import cats.effect.kernel.Resource
+import cats.effect.kernel.Sync
+
+abstract class MutationObserver[F[_]] private[dom] {
+
+  def observe(target: Node[F], options: dom.MutationObserverInit): F[Unit]
```

Comment:
Shall we also provide a wrapper for `dom.MutationObserverInit`?

Reply:
And `dom.ResizeObserverOptions`?

Reply:
Good question :) ok, so usual disclaimer about arbitrariness ...

I've been trying to just reuse these `*Init` and `*Options` types whenever possible, so that we don't have to wrap _everything._ They are mutable, which is not great, but I think their usual usage pattern is as builders local to the method call, rather than passing them all over the place. So probably fine. And all their member types are simple stuff (strings, booleans, enums). So IMO the trade-off is not worth it: if we wrapped it, it would be more annoying for us to maintain, and more annoying for the user to use, so lose-lose.